pkg/registry/apis/provisioning/jobs/migrate/legacy_resources.go

```diff
@@ -70,17 +71,29 @@ func (m *legacyResourcesMigrator) Migrate(ctx context.Context, rw repository.Rea
 	}
 
 	progress.SetMessage(ctx, "migrate folders from SQL")
-	if err := m.folderMigrator.Migrate(ctx, namespace, repositoryResources, progress); err != nil {
-		return fmt.Errorf("migrate folders from SQL: %w", err)
+	if m.clients != nil {
+		clients, err := m.clients.Clients(ctx, namespace)
+		if err != nil {
+			return err
+		}
+		folderClient, err := clients.Folder()
+		if err != nil {
+			return err
+		}
+		err = export.ExportFolders(ctx, rw.Config().Name, provisioning.ExportJobOptions{}, folderClient, repositoryResources, progress)
```

Comment:
This whole topic with the dual writer makes me dizzy. 

Correct me if I am wrong. 

With this we now read the folders from the API Server which are stored in SQL because we don't care about their history.

Is that the case?

Reply:
> we don't care about their history.

we don't maintain history for folders in legacy, so we are not losing anything